orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/WaitForDisabledServerGroupTask.java

```diff
@@ -64,6 +69,23 @@ public TaskResult execute(@NotNull StageExecution stage) {
     }
   }
 
+  private boolean isPartialDisable(Object desiredPercentage) {
+    if (desiredPercentage == null) {
+      return false;
+    }
+
+    if (desiredPercentage instanceof Integer) {
+      return ((Integer) desiredPercentage) < 100;
+    }
+
+    if (desiredPercentage instanceof String) {
+      return Integer.parseInt((String) desiredPercentage) < 100;
+    }
+
+    log.warn("Unexpected value in stage context, desiredPercentage={}", desiredPercentage);
+    return true;
```

Comment:
`stage.mapTo()` (and a try/catch) would probably save these conversations. 

Should the default be a partial disable if there's something wonky in `desiredPercentage`? That doesn't seem right.

Reply:
ah, I always assumed `stage.mapTo()` was Groovy shenanigans

The default sounds wonky but it really means "when in doubt, don't enforce anything"

Reply:
in f9a1dadbe717be6b35e9ac52d702d643bcfe6988